src/protocol/secureRequests.ts

```diff
@@ -0,0 +1,389 @@
+  data: Buffer,
+  requestType: LatticeSecureEncryptedRequestType
+): Promise<Buffer> {
+  // Generate a random message id for internal tracking
+  // of this specific request (internal on both sides).
+  const msgId = randomBytes(4);
+  // Serialize the request data into encrypted request
+  // payload data.
+  const payloadData = serializeSecureRequestEncryptedPayloadData(
+    client,
+    data,
+    requestType
+  );
+  // Serialize the payload data into an encrypted secure
+  // request message.
+  const msg = serializeSecureRequestMsg(
+    client,
+    msgId,
+    LatticeSecureMsgType.encrypted,
+    payloadData
+  );
+  // Send request to Lattice
+  const resp = await request({ 
+    url: client.url, 
+    payload: msg 
+  });
+  // Deserialize the response payload data
+  const encRespPayloadData = deserializeResponseMsgPayloadData(
+    client,
+    msgId,
+    resp
+  );
+  // Return decrypted response payload data
+  return decryptEncryptedLatticeResponseData(
+    client, 
+    encRespPayloadData,
+    requestType
+  );
+}
+
+/**
+ * @internal
+ * Serialize a Secure Request message for the Lattice.
+ * All outgoing SDK requests are of this form.
+ * @param client - Instance of the Client
+ * @param msgId - Random 4 bytes of data for internally tracking this message
+ * @param secureRequestType - 0x01 for connect, 0x02 for encrypted
+ * @param payloadData - Request data
+ * @return {Buffer} Serialized message to be sent to Lattice
+ */
+function serializeSecureRequestMsg(
+  client: Client,
+  msgId: Buffer,
+  secureRequestType: LatticeSecureMsgType,
+  payloadData: Buffer,
+): Buffer {
+  // Sanity check request data
+  if (msgId.length !== 4) {
+    throw new Error('msgId must be four bytes');
+  }
+  const isValidRequestType = (
+    secureRequestType === LatticeSecureMsgType.connect ||
+    secureRequestType === LatticeSecureMsgType.encrypted
+  );
+  const isValidConnectPayloadDataSz = (
+    payloadData.length === szs.connect.request.data  
+  );
+  const isValidEncryptedPayloadDataSz = (
+    payloadData.length === szs.requestPayloadData  
+  );
+  const isValidPayloadDataSz = (
+    isValidConnectPayloadDataSz || isValidEncryptedPayloadDataSz
+  ); 
+  if (!isValidRequestType) {
+    throw new Error('Invalid Lattice secure request type');
+  } else if (!isValidPayloadDataSz) {
+    throw new Error('Invalid Lattice secure request payload size');
+  }
+
+  // Build payload
+  let payload;
+  const _payload = Buffer.alloc(szs.requestPayload);
+  payload[0] = secureRequestType;
+  payloadData.copy(payload, 1);
+
+  // Encrypt if needed
+  if (isValidEncryptedPayloadDataSz) {
+    payload = aes256_encrypt(_payload, client.sharedSecret);
+  } else {
+    payload = _payload;
+  }
+
+  // Construct the object first
+  const req: LatticeSecureRequest = {
+    protocolVersion: LatticeProtocolVersion.v1,
+    msgType: LatticeMsgType.secure,
+    msgId,
+    payloadLen: payload.length,
+    payload,
+    checksum: checksum(payload),
```

Comment:
Running e2e tests will sniff this out, but I actually think the checksum is only included in encrypted requests... specifically in the data prior to encrypting...

The only other checksum in firmware seems to come from the wire protocol, which is constructed by the GCE, not here...